src/test/scala/kanaloa/reactive/dispatcher/DispatcherSpec.scala

```diff
@@ -32,6 +37,39 @@ class DispatcherSpec extends SpecWithActorSystem {
     }
   }
 
+  "pushing work dispatcher" should {
+    trait SimplePushingDispatchScope extends ScopeWithActor {
+      val dispatcher = system.actorOf(PushingDispatcher.props(
+        name = "test",
+        Backend((i: String) ⇒ Future.successful(MessageProcessed(i)))
+      )(ResultChecker.simple[MessageProcessed]))
+    }
+    "work happily with simpleBackend" in new SimplePushingDispatchScope {
+
+      dispatcher ! "3"
+      expectMsg(MessageProcessed("3"))
+
+    }
+
+    "let simple backend reject unrecognized message" in new SimplePushingDispatchScope {
+      dispatcher ! 3
+      expectMsgType[WorkFailed]
+
+    }
+
+    "let simple result check fail on  unrecognized reply message" in new ScopeWithActor {
```

Comment:
minor issues -- there's an extra space between `on` and `unrecognized`, and the newlines are kinda inconsistent